backend/de.metas.handlingunits.base/src/main/java/de/metas/handlingunits/inventory/InventoryService.java

```diff
@@ -261,6 +269,9 @@ public HuId createInventoryForMissingQty(@NonNull final CreateVirtualInventoryWi
 
 		final InventoryId inventoryId = createInventoryHeader(createHeaderRequest).getId();
 
+		final Set<FlatrateTermId> contractIds = modularContractProvider.streamPurchaseContractsForSalesOrderLine(req.getOrderAndLineId())
```

Comment:
why not simply providing the contractId as parameter (via InventoryLineCreateRequest) 
... instead of coupling this InventoryService with ModularContractProvider
?